### PR TITLE
Fix validation of type for enttities in subscription

### LINKED ISSUE #1964: [BUG] in POST v2/subscriptions, in entities/type field not plain ascii char are allowed
in POST v2/subscriptions, in entities/type field not plain ascii char are allowed.

#### Dataset
```
      | type       |
      |------------|
      | habitación |
      | españa     |
      | barça      |
```

#### subscription request
```
POST http://localhost:1026/v2/subscriptions
Content-Type: application/json
Fiware-Service: test_entities_type
Fiware-ServicePath: /test
{"notification": {"callback": "http://localhost:1234", "attributes": ["temperture_0"]}, "expires": "2016-04-05T14:00:00.00Z", "subject": {"entities": [{"idPattern": ".*", "type": "barça"}], "condition": {"attributes": ["temperature_0"]}}}
```

#### subscription response
```
http code: 201
date: Tue, 29 Mar 2016 16:03:53 GMT
connection: Keep-Alive
content-length: 0
location: /v2/subscriptions/56faa7696f3a7e103a8e5569
```

#### doc in mongo
```
{
	"reference": "http://localhost:1234",
	"throttling": 0 L,
	"expression": {
		"q": "",
		"geometry": "",
		"georel": "",
		"coords": ""
	},
	"format": "JSON",
	"entities": [{
		"type": "bar\xe7a",
		"id": ".*",
		"isPattern": "true"
	}],
	"expiration": 1459864800 L,
	"servicePath": "/test",
	"_id": ObjectId("56faa7696f3a7e103a8e5569"),
	"conditions": [{
		"type": "ONCHANGE",
		"value": ["temperature_0"]
	}],
	"attrs": ["temperture_0"]
}
```

#### expected response
```
http code: 400
{"error":"BadRequest","description":"Invalid characters in entity type"}
```

### PR DESCRIPTION
**Pull request by crbrox**
@fgalan 
@kzangeli 

hopefully closes #1964 #1965 #1967

### CODE REVIEW

File: CHANGES_NEXT_RELEASE [hunk 1]
```diff
@@ -0,0 +1,3 @@
+=======
```
Comment (fgalan) [inline CHANGES_NEXT_RELEASE]: `====` is probably a leftover.

File: src/lib/jsonParseV2/parseSubscription.cpp [hunk 2]
```diff
@@ -319,6 +320,27 @@ static std::string parseEntitiesVector(ConnectionInfo* ciP, EntityIdVector* eivP
 
         return oe.render(ciP, "");
       }
+      size_t len = type.size();
+      if (len > MAX_ID_LEN)
```
Comment (fgalan) [inline src/lib/jsonParseV2/parseSubscription.cpp]: This code is used to parse also in the case of update subscription information (PATCH method)?

**Comment by crbrox** [comment]
To be resumed after more important fixes (#1947, #1972)  and likely it will  include all bugs about validations in subscription fields. 

(Do not delete the branch, please)

**Comment by fgalan** [comment]
The validations included in this PR has been already fixed in some other PRs (note that the #1964, #1965, #1967 has been closed, after checking by QA). Thus, once "saved" the .test included in bug/1964_validation_type_subs branch (see PR https://github.com/telefonicaid/fiware-orion/pull/2887) it is save to delete the branch.

**Pull request closed without merging**